NLog.Windows.Forms/RichTextBoxTarget.cs

```diff
@@ -202,71 +235,231 @@ public RichTextBoxTarget()
         public RichTextBox TargetRichTextBox { get; set; }
 
         /// <summary>
-        /// Form created (true) or used an existing (false). Set after <see cref="InitializeTarget"/>
+        /// Form created (true) or used an existing (false). Set after <see cref="InitializeTarget"/>. Can be true only if <see cref="AllowAccessoryFormCreation"/> is set to true (default).
         /// </summary>
         public bool CreatedForm { get; set; }
 
         /// <summary>
+        /// Gets or sets a value indicating whether to create accessory form if the specified form/control combination was not found during target initialization.
+        /// </summary>
+        /// <remarks>
+        /// If set to false and the control was not found during target initialiation, the target would skip events until the control is found during <see cref="ReInitializeAllTextboxes"/> call
+        /// </remarks>
+        /// <docgen category='Form Options' order='10' />
+        [DefaultValue(true)]
+        public bool AllowAccessoryFormCreation { get; set; }
+
+        /// <summary>
+        /// gets or sets the 
+        /// </summary>
+        /// <remarks>
+        /// </remarks>
+        /// <docgen category='Form Options' order='10' />
+        [DefaultValue(RichTextBoxTargetMessageRetentionStrategy.None)]
+        public RichTextBoxTargetMessageRetentionStrategy MessageRetention 
+        { 
+            get { return messageRetention; } 
+            set 
+            {
+                lock (messageQueueLock)
```

Comment:
I don't think so. I think all modifications to queue are to be synchronized. Anyway, the scope looks large because of brackets mostly )